bin/common.ml

```diff
@@ -763,6 +764,33 @@ module Builder = struct
                ])
             Automatic
         & info [ "file-watcher" ] ~doc)
+    and+ watch_exclusions =
+      let doc_escape s =
+        let escape_char = function
+          | '$' -> "\\$"
+          | c -> Char.escaped c
+        in
+        String.to_seq s |> List.of_seq |> List.map ~f:escape_char
+        |> String.concat ~sep:""
```

Comment:
What is the purpose of this escaping?

Also, I know that it doesn't matter here, but converting a string to a list and back is famously inefficient; using a `Buffer.t` would be much better.

Reply:
> What is the purpose of this escaping?
> 
> Also, I know that it doesn't matter here, but converting a string to a list and back is famously inefficient; using a `Buffer.t` would be much better.

Cmdliner documentation markup language requires escaping `$` and `\`: https://erratique.ch/software/cmdliner/doc/tool_man.html#doclang . I've added that in the doc comment as well.

And yeah, switched to `Buffer.t`.